### ISSUE TITLE
Bug: Import Status - STAC validation errors not reported

### ISSUE THREAD

**Issue by billgeo** [linz/geospatial-data-lake#549]
## Bug Description

I've noticed some JSON/STAC errors are not reported to the end user via the import-status endpoint. Can we ensure **all** errors are reported to the end user in some way?

e.g. When I try and import data with incorrect key like 'checksum:multihash' instead of 'checksum:file' I get this error in the logs. 

```
[...]
	2021-04-14T10:23:34.853+12:00	File "/src/task/utils.py", line 124, in validate
	2021-04-14T10:23:34.853+12:00	asset_dict = {"url": asset_url, "multihash": asset["file:checksum"]}
	2021-04-14T10:23:34.853+12:00	KeyError: 'file:checksum'
```

But the import status looks lie this:

```
{
  "statusCode": 200,
  "body": {
    "step function": {
      "status": "FAILED"
    },
    "validation": {
      "status": "Pending",
      "errors": []
    },
    "metadata upload": {
      "status": "Pending",
      "errors": []
    },
    "asset upload": {
      "status": "Pending",
      "errors": []
    }
  }
}
```

No way for the end user to find out what is going on (beside asking someone int he product team to look in the logs.

Example STEP Funciton instance
https://ap-southeast-2.console.aws.amazon.com/states/home?region=ap-southeast-2#/executions/details/arn:aws:states:ap-southeast-2:632223577832:execution:nonproddatasetversioncreation8597518E-i9JA9Y04AIGV:a0e841439ca611eb84b027099ebdcf03

## How to Reproduce

 

1. Do …
1. Run `…`

What did you expect to happen? 

What actually happened? 

## Software Context

Operating system: 

Environment: 

Relevant software versions:

- AWS CLI: 
- Poetry:

**Comment by billgeo** [comment]
Could be resolved as part of #517

**Comment by billgeo** [comment]
Closing this in favour of #517  (added an acceptance criteria to cover this)
